src/Options/CMakeLists.txt

```diff
@@ -18,8 +24,9 @@ spectre_target_headers(
 
 target_link_libraries(
   ${LIBRARY}
-  INTERFACE
+  PUBLIC
   ErrorHandling
   Utilities
```

Comment:
I still haven't understood exactly what the different types of dependencies mean, but why is Utilities in this section?

Reply:
All libs that a target needs to compile must be in `PRIVATE`, i.e. also all headers that are included from all the cpp files. All libs included from _headers_ in the target must be in `INTERFACE`. Libs that appear in both must be in `PUBLIC`.
Examples:
- A cpp file includes its header and the header includes `Utilities` => `Utilities` ist `PUBLIC`
- A cpp file includes `Utilities` but none of the headers include `Utilities` => `Utilities` is `PRIVATE`
- Only a header that is included by none of the cpp files includes `Utilities` => `Utilities` is `INTERFACE`